lib/ansible/modules/windows/win_package.ps1

```diff
@@ -221,12 +227,21 @@ Function Get-ProgramMetadata($state, $path, $product_id, $credential, $creates_p
             } catch {
                 Fail-Json -obj $result -message "failed to get product_id from MSI at $($path): $($_.Exception.Message)"
             }
-        } elseif ($creates_path -eq $null -and $creates_service -eq $null) {
-            # we need to fail without the product id at this point
-            Fail-Json $result "product_id is required when the path is not an MSI or the path is an MSI but not local"
+        } elseif ($creates_path -eq $null -and $creates_service -eq $null -and $display_name -eq $null) {
+            # we need to fail without the product id or display_name at this point
+            Fail-Json $result "product_id or display_name is required when the path is not an MSI or the path is an MSI but not local"
         }
     }
 
+    if ($display_name -ne $null) {
```

Comment:
You may as well set this when `$metadata` is created, the value will still be `$null` when `$display_name/version` is also null